paseos/activities/activity_processor.py

```diff
@@ -0,0 +1,110 @@
+import asyncio
+from contextlib import suppress
+from timeit import default_timer as timer
+
+from loguru import logger
+
+from paseos.activities.activity_runner import ActivityRunner
+
+
+class ActivityProcessor:
+    """This class specifies the processor of paseos running in the background during an activity."""
+
+    def __init__(
+        self,
+        update_interval: float,
+        power_consumption_in_watt: float,
+        paseos_instance,
+        activity_runner: ActivityRunner,
```

Comment:
If we want only one of these objects, this should be a list

Reply:
I think for now I would limit users to just one plain activity and they need to do something like

```Python
async def someParallelActivity(args):
       # Call one (potentially async) thing
        await some_function()

        # Do something else
        await some_other_function()

```

Reply:
Ok, I agree. This is something that we must explain to users in a pedagogical way eventually.

Reply:
Added to #6